### ISSUE TITLE
Nested Deep Links

### ISSUE THREAD

**Issue by blakelee** [raamcosta/compose-destinations#153]
Hey, I've been playing around with your library for a couple weeks now and I'm trying to test out a few use case's before I can try and convince my company to adopt it.

I am trying to deep link into a nested nav graph but it looks like that a nested deep link doesn't use its DestinationsNavHost when it's called.

For example here is my setup. I have two DestinationsNavHost's (DNH): main and sample.
<img width="541" alt="image" src="https://user-images.githubusercontent.com/9651986/172308314-d4f8a729-5878-4ba2-ae96-115d0a9acfd9.png">
                                                  
I have also color coded each screen and DNH so that it's easy to see what is holding which content.
Black border = main DNH
Blue border = sample DNH

In my video, you can see that if I navigate from a deep link outside of the app then it will render the screen without the blue border which means the sample DNH is not being shown. The same is true if I open the deep link from ScreenB since ScreenB is part of the main DNH but not the sample DNH. ScreenB deep links into ScreenE but doesn't show the border from ScreenC's DNH.

If I open the deep link while in ScreenD, then ScreenE gets shown properly within the sample DNH from ScreenC.

Is this simply a limitation of nested deep linking or can it be solved?

https://user-images.githubusercontent.com/9651986/172306807-6ebfca4b-e89a-4fb5-916a-ded15aa49b09.mp4

**Comment by raamcosta** [comment]
Hey @blakelee 👋 

Just a quick question, have you tried this with vanilla compose navigation? 

Also, is this sample project a public repo I can look at?

**Comment by blakelee** [comment]
Here is the sample https://github.com/blakelee/Compose-Destinations-Sample

I have no tried vanilla compose navigation. I can probably migrate to it real quick to test it.

**Comment by raamcosta** [comment]
If you want I can do that as well. It’s not the first or the last time I’ll do it 😅
I have practiced hands in that

**Comment by blakelee** [comment]
Sure, if it's not an issue for you. I was essentially just copying+pasting the code from here and modifying it https://android-review.googlesource.com/c/platform/frameworks/support/+/1818835/6/navigation/navigation-compose/samples/src/main/java/androidx/navigation/compose/samples/NavigationSamples.kt#138

**Comment by blakelee** [comment]
I might have figured it out. I noticed that in the vanilla example that there's not place for a modifier in the nested navigation block. Then I removed ScreenC from my project completely and changed the BottomBarItem to be ScreenA and ScreenB which are destinations then the 3rd item I swapped out ScreenC with my nested NavGraph `NavGraph.sample`.

I will push my code to show you. It looks like there's only supposed to be one DestinationsNavHost per set of graphs including a graph's nested graphs.

I tried to solve this over Saturday and Sunday and I couldn't figure it out but now it looks like it's working :)
